src/api/app/models/bs_request_action.rb

```diff
@@ -962,6 +966,22 @@ def check_for_expand_errors!(add_revision)
     end
   end
 
+  private
+
+  def set_associations
```

Comment:
Very minor comment and up to you if you want to change this but I would say that `set_associations` is too vague, I would prefer something like `set_source_and_target_associations`